### PR TITLE
Updated socket_impl.cpp to fix Bad File Descriptor Error

### PR DESCRIPTION
**Pull request by stkrwork**
modified:   source/corvusoft/restbed/detail/socket_impl.cpp

### CODE REVIEW

**Comment by ben-crowhurst** [comment]
@stkrwork Your PR is very hard to understand, it seems your local branch has clobbered all the formatting/whitespace.  Can you run the pre-commit hook on it?

**Comment by stkrwork** [comment]
Added a new commit that highlights the changes i made. This new commit also respects the connection timeout counter. This counter will not be started if the connection isn't open.

**Comment by ben-crowhurst** [comment]
I'm still struggling to digest this PR, there is far too much static involved around whitespace.  Please inform me of the changes and i'll crank them out by hand.  Alternatively install the GIT pre-commit hook and format the code to reduce the clutter.  See [STANDARDS.md](https://github.com/Corvusoft/restbed/blob/master/documentation/STANDARDS.md) for further details.

We really appreciate the effort you've put into this and are very keen getting is merged ASAP.

File: source/corvusoft/restbed/detail/socket_impl.cpp [hunk 1]
```diff
@@ -15,7 +15,6 @@
 #include <asio/connect.hpp>
 #include <asio/read_until.hpp>
 
-
```
Comment (ben-crowhurst) [inline source/corvusoft/restbed/detail/socket_impl.cpp]: static

File: source/corvusoft/restbed/detail/socket_impl.cpp [hunk 2]
```diff
@@ -38,468 +37,482 @@ using asio::io_service;
 using asio::steady_timer;
 
 #ifdef BUILD_SSL
-    using asio::ssl::stream;
```
Comment (ben-crowhurst) [inline source/corvusoft/restbed/detail/socket_impl.cpp]: static

File: source/corvusoft/restbed/detail/socket_impl.cpp [hunk 3]
```diff
@@ -38,468 +37,482 @@ using asio::io_service;
...
+using asio::ssl::stream;
 #endif
 
 namespace restbed
 {
-    namespace detail
```
Comment (ben-crowhurst) [inline source/corvusoft/restbed/detail/socket_impl.cpp]: static, etc... you get the point.

File: source/corvusoft/restbed/detail/socket_impl.cpp [hunk 4]
```diff
@@ -38,468 +37,482 @@ using asio::io_service;
...
-    {
-        SocketImpl::SocketImpl( const shared_ptr< tcp::socket >& socket, const shared_ptr< Logger >& logger ) : m_is_open( socket->is_open( ) ),
-            m_buffer( nullptr ),
-            m_logger( logger ),
-            m_timeout( 0 ),
-            m_timer( make_shared< asio::steady_timer >( socket->get_io_service( ) ) ),
-            m_strand( make_shared< io_service::strand > ( socket->get_io_service( ) ) ),
-            m_resolver( nullptr ),
-            m_socket( socket )
```
Comment (ben-crowhurst) [inline source/corvusoft/restbed/detail/socket_impl.cpp]: static, etc... you get the point.

File: source/corvusoft/restbed/detail/socket_impl.cpp [hunk 5]
```diff
@@ -38,468 +37,482 @@ using asio::io_service;
...
-                        const auto data_ptr = asio::buffer_cast< const Byte* >( data->data( ) );
-                        success( Bytes( data_ptr, data_ptr + length ) );
-                    }
-                } );
-#ifdef BUILD_SSL
-            }
-            else
-            {
-                auto data = make_shared< asio::streambuf >( );
-                asio::async_read( *m_ssl_socket, *data, asio::transfer_exactly( length ), [ this, data, success, failure ]( const error_code code, const size_t length )
```
Comment (ben-crowhurst) [inline source/corvusoft/restbed/detail/socket_impl.cpp]: How are we meant to see whats changed in this file?

**Comment by stkrwork** [comment]
i'll revert it and code it in again with the hook from the beginning. hopefully this will fix it and make it more readable.

**Pull request closed without merging**